joblib/hashing.py

```diff
@@ -241,6 +241,53 @@ def save(self, obj):
         Hasher.save(self, obj)
 
 
+class PandasHasher(NumpyHasher):
+    """ Special case the hasher for when pandas is loaded.
+    """
+
+    def __init__(self, hash_name='md5', coerce_mmap=False):
+        """
+            Parameters
+            ----------
+            hash_name: string
+                The hash algorithm to be used
+            coerce_mmap: boolean
+                Make no difference between np.memmap and np.ndarray
+                objects.
+        """
+        NumpyHasher.__init__(
+            self, hash_name=hash_name, coerce_mmap=coerce_mmap
+        )
+        # delayed import of pandas, to avoid tight coupling
+        import pandas as pd
+        self.pd = pd
+
+    def save(self, obj):
+        """ Subclass the save method, to hash pandas objects using fast Pandas
+            implementations rather than pickling them. Of course, this is a
+            total abuse of the Pickler class.
+        """
+        if isinstance(obj, self.pd.DataFrame):
+            buf = io.BytesIO()
```

Comment:
Using a buffer creates a copy and is thus prohibitive. The best possible alternative would be to use pickle's protocol 5 (described by [PEP 574](https://peps.python.org/pep-0574/)) with a custom File Object (i.e. a class implementing the `write` and `close` methods) to avoid any memory copy. Such a File Object has to subclass the `Pickler` class. This makes the contribution slightly harder.